test/integration/mpi_job_controller_test.go

```diff
@@ -822,6 +823,96 @@ func TestMPIJobWithVolcanoScheduler(t *testing.T) {
 	}
 }
 
+func TestMPIJobManagedExternally(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	s := newTestSetup(ctx, t)
+	startController(ctx, s.kClient, s.mpiClient, nil)
+
+	mpiJob := &kubeflow.MPIJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "job",
+			Namespace: s.namespace,
+		},
+		Spec: kubeflow.MPIJobSpec{
+			SlotsPerWorker: ptr.To[int32](1),
+			RunPolicy: kubeflow.RunPolicy{
+				CleanPodPolicy: ptr.To(kubeflow.CleanPodPolicyRunning),
+				ManagedBy:      ptr.To(kubeflow.MultiKueueController),
+			},
+			MPIReplicaSpecs: map[kubeflow.MPIReplicaType]*kubeflow.ReplicaSpec{
+				kubeflow.MPIReplicaTypeLauncher: {
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "main",
+									Image: "mpi-image",
+								},
+							},
+						},
+					},
+				},
+				kubeflow.MPIReplicaTypeWorker: {
+					Replicas: ptr.To[int32](2),
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "main",
+									Image: "mpi-image",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// 1. The job must be created
+	var err error
+	mpiJob, err = s.mpiClient.KubeflowV2beta1().MPIJobs(s.namespace).Create(ctx, mpiJob, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed sending job to apiserver: %v", err)
+	}
+	backoff := wait.Backoff{
+		Duration: moderateTimeout,
+		Factor:   1.0,
+		Steps:    3,
+	}
+	err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+		// 2. Status is not getting updated
+		mpiJob = validateMPIJobStatus(ctx, t, s.mpiClient, mpiJob, nil)
+		if mpiJob.Status.StartTime != nil {
+			t.Errorf("MPIJob should be missing startTime")
+		}
+		// 3. There should be no conditions, even the one for create
+		if mpiJobHasCondition(mpiJob, kubeflow.JobCreated) {
+			t.Errorf("MPIJob shouldn't have any condition")
+		}
+		// 4. No Pods or Services created
+		pods, err := getPodsForJob(ctx, s.kClient, mpiJob)
+		if err != nil {
+			t.Fatalf("Failed getting pods for the job: %v", err)
+		}
```

Comment:
Could we verify if the Launcher Job doesn't exist as well since the MPIOperator will create the Launcher batch/v1 Job and Worker Pods?

Reply:
done